tests/unittests/lib.arithmetic.art

```diff
@@ -1116,6 +1116,188 @@ do [
 ]
 
 
+topic "mod"
+do [
+    
+    topic "mod - :integer :integer"
+    
+    ensure -> 0 = 1 % 1
+    ensure -> 0 = 0 % 1
+    ensure -> 0 = mod 1 1
+    ensure -> 0 = mod 0 1
+    passed
+    
+    try? -> 1 % 0
+    else -> passed
+    
+    try? -> mod 1 0
+    else -> passed
+    
+    ensure -> 1 = 1 % 2
+    ensure -> 0 = 2 % 1
+    ensure -> 1 = mod 1 2
+    ensure -> 0 = mod 2 1
+    passed
+    
+    ensure -> 0 = 8 % neg 4
+    passed
+    
+    a: 8
+    mod 'a 4
+    ensure -> a = 0
+    a: 8
+    'a % 3
+    ensure -> a = 2
+    passed
+    
+    topic "mod - :int :floating"
+    
+    ensure -> 0 = 1 % 1.0
+    ensure -> 0 = 0 % 1.0
+    ensure -> 0 = mod 1 1.0
+    ensure -> 0 = mod 0 1.0
+    passed
+
+    ensure -> (1 % 1.0) = 1.0 % 1
+    ensure -> (0 % 1.0) = 0.0 % 1
+    ensure -> (mod 1 1.0) = mod 1.0 1
+    ensure -> (mod 0 1.0) = mod 0.0 1
+    passed
+    
+    try? -> 1 % 0.0
+    else -> passed
+
+    try? -> 1.0 % 0
+    else -> passed
+    
+    try? -> mod 1.0 0
+    else -> passed
+
+    try? -> mod 1.0 0
+    else -> passed
+    
+    ensure -> 0 = 1 % 0.5
+    ensure -> 1 = 1.0 % 2
+    ensure -> 0 = 8 % 4.0
+    ensure -> 0 = mod 1 0.5
+    ensure -> 1 = mod 1.0 2
+    ensure -> 0 = mod 8 4.0
+    passed
+    
+    ensure -> (8 % 4.0) = 8.0 % 4
+    ensure -> (mod 8 4.0) = mod 8.0 4
+    passed
+    
+    ensure -> 0 = 8 % neg 4.0
+    passed
+
+    ensure -> (8 % neg 4.0) = 8.0 % neg 4
+    passed
+    
+    a: 8
+    mod 'a 4.0
+    ensure -> a = 0.0
+    a: 8
+    'a % 3
+    ensure -> a = 2.0
+    passed
+    
+    
+    ; topic "mod - :int :rational"
```

Comment:
How does should it work?

![image](https://user-images.githubusercontent.com/78623871/225948365-840e210e-218a-470b-93e4-ce6ee738477c.png)
Is it right?

Reply:
That's a difficult one. I guess I'm not modulo-ing rational numbers very often... lol 

The first (`1/2 % 2 => 1/2`) looks correct.
The second one (`3/5 % 2 => 3/5`) also looks correct.

Now, the last one (`1/2 % 1/2 => 1/10`) looks bizarre and wrong to say the least. It must have to do with Rational numbers being converted to floating-point values internally before the operation takes place?

I'll have to investigate... Hmm...